### PR TITLE
Epic spells foundation

### PR DESCRIPTION
**Pull request by zanderriggs**
Just taking little bits off this. 

Added the spell seed data from the original site with some cleanup. Threw it in the page for a quick visualization for myself.

Still not super confident with ts-- just need more time messing around.

### CODE REVIEW

File: src/data/epic-spell-seed.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        spellresist: 'yes',
+    },
+    {
+        id: 23,
+        name: 'transform',
+        dc: 21,
+        components: ['v', 's'],
+        casting: 60,
+        range: '300 feet',
+        target: '	One creature or inanimate, nonmagical object',
```
Comment (mikerapin) [inline src/data/epic-spell-seed.ts:282]:
```suggestion
        target: 'One creature or inanimate, nonmagical object',
```

Nit pick :P

File: src/pages/epic-spells.tsx [hunk 2]
```diff
@@ -12,6 +13,18 @@ export const EpicSpells = () => {
                 </div>
             </header>
             <section>Create an epic spell.</section>
+            <div>
+                <ul>
+                    {SPELL_SEEDS.map(seed => (
+                        <div key={seed.id}>
+                            <span>
+                                <input type="checkbox" id={seed.id.toString()} name={seed.name + '-checkbox'}></input>
```
Comment (mikerapin) [inline src/pages/epic-spells.tsx:21]: > `id={seed.id.toString()`

Probably not needed to convert the number to a string here.

> name={seed.name + '-checkbox'}

I would also use a [string template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for the name just for cleanliness:

```
name={`${seed.name}-checkbox`}
```

File: src/pages/index.tsx [hunk 3]
```diff
@@ -39,10 +39,10 @@ const links = [
...
     {
-        text: 'XP Table',
-        url: '/xp',
+        text: 'Epic Spells Things',
+        url: '/epic',
         badge: true,
-        description: "Here's the 5e XP table because no one ever knows these numbers",
+        description: "Knowing magic is cool This'll make you cooler.",
         color: '#1099A8',
     },
```
Comment (mikerapin) [inline src/pages/index.tsx:47]: Are we getting rid of the XP table?

File: src/pages/index.tsx [hunk 4]
```diff
@@ -39,10 +39,10 @@ const links = [
...
     {
-        text: 'XP Table',
-        url: '/xp',
+        text: 'Epic Spells Things',
+        url: '/epic',
         badge: true,
-        description: "Here's the 5e XP table because no one ever knows these numbers",
+        description: "Knowing magic is cool This'll make you cooler.",
```
Comment (mikerapin) [inline src/pages/index.tsx:45]: hell yeah

**Review by mikerapin — CHANGES_REQUESTED** [review]

File: src/pages/epic-spells.tsx (hunk 2 above)
Comment (zanderriggs) [reply src/pages/epic-spells.tsx:21]: I was trying to figure out the string template literal and just couldn't make it work. I didn't have enough '{' in there.

I thought I didn't need to make the id a string either, but the id property on an input only takes a string and the id on the model is a number. I'll just make the id a string if that makes more sense. I didn't have a specific reason for making it a number to begin with.

File: src/pages/index.tsx (hunk 3 above)
Comment (zanderriggs) [reply src/pages/index.tsx:47]: I guess I preemptively pulled this. We had talked briefly about narrowing scope of things and I think past Zander just pulled this based on that conversation. The page still exists. I was just removing the link. We can keep it 👍

File: src/data/epic-spell-seed.ts (hunk 1 above)
Comment (zanderriggs) [reply src/data/epic-spell-seed.ts]: Oops. Thanks for catching this.

File: src/pages/index.tsx (hunk 3 above)
Comment (zanderriggs) [reply src/pages/index.tsx]: b5e8152

File: src/pages/epic-spells.tsx (hunk 2 above)
Comment (zanderriggs) [reply src/pages/epic-spells.tsx]: ae4602f

File: src/pages/index.tsx (hunk 4 above)
Comment (zanderriggs) [reply src/pages/index.tsx]: I did miss punctuation in there, so I fixed it ae4602f

File: src/pages/epic-spells.tsx [hunk 5]
```diff
@@ -12,6 +13,18 @@ export const EpicSpells = () => {
                 </div>
             </header>
             <section>Create an epic spell.</section>
+            <div>
+                <ul>
+                    {SPELL_SEEDS.map(seed => (
+                        <div key={seed.id}>
+                            <span>
+                                <input type="checkbox" id={seed.id} name={'${seed.name}-checkbox'}></input>
```
Comment (mikerapin) [inline src/pages/epic-spells.tsx:21]:
```suggestion
                                <input type="checkbox" id={seed.id} name={`${seed.name}-checkbox`}></input>
```

Backticks!

**Review by mikerapin — CHANGES_REQUESTED** [review]

File: src/pages/epic-spells.tsx (hunk 5 above)
Comment (zanderriggs) [reply src/pages/epic-spells.tsx:21]: Backticks?? okay okay 😓

**Review by mikerapin — APPROVED** [review]

**Pull request merged**